tasking/pulp/tasking/storage.py

```diff
@@ -0,0 +1,138 @@
+import errno
+import os
+import shutil
+import stat
+
+from celery import task
+from django.conf import settings as pulp_settings
+
+
+def _working_dir_root(worker_name):
+    """Return the path to the working directory of a worker.
+
+    :param worker_name:     Name of worker for which path is requested
+    :type  worker_name:     basestring
+
+    :returns: working_directory setting from server config
+    :rtype:   str
+    """
+    return os.path.join(pulp_settings.SERVER['working_directory'], worker_name)
+
+
+def create_worker_working_directory(worker_name):
+    """Create a working directory for a worker.
+
+    Create a working directory inside the working_directory as specified in the setting file.
+
+    :param worker_name:     Name of worker that uses the working directory created
+    :type  worker_name:     basestring
+
+    :returns: Path to the working directory for a worker
+    :rtype:   str
+    """
+    working_dir_root = _working_dir_root(worker_name)
+    os.mkdir(working_dir_root)
+    return working_dir_root
+
+
+def delete_worker_working_directory(worker_name):
+    """Delete worker's working directory.
+
+    Delete a working directory inside the working_directory as specified in the setting file.
+
+    :param worker_name:     Name of worker that uses the working directory being deleted
+    :type  worker_name:     basestring
+    """
+    _rmtree(_working_dir_root(worker_name))
+
+
+def _working_directory_path():
+    """Get path of task working directory.
+
+    Get the path for a task working directory inside a workers working directory.
+
+    :returns: full path on disk to the working directory for current task
+    :rtype:   basestring
+    """
+    current_task = task.current
+    task_id = current_task.request.id
+    worker_name = current_task.request.hostname
+    if current_task and current_task.request and task_id and worker_name:
+        return os.path.join(_working_dir_root(worker_name), task_id)
+
+
+def delete_working_directory():
+    """Delete working directory for a particular task."""
+    working_dir_root = _working_directory_path()
+
+    if working_dir_root:
```

Comment:
I believe it would be better to not check and instead try to delete and suppress exceptions.